### DISCUSSION TITLE
Vídeo 05 - Pergunta adicional

### DISCUSSION THREAD

**Discussion by thpoiani** [thpoiani/lab-quarkus#7] [Q&A]
Com a configuração atual, **secrets**, como usuário e senha do banco de dados, são passados diretamente para o `docker-compose.yml`.

Gostaria de receber sugestões de outras abordagens para aumentar a segurança desse sistema.

<details>
  <summary>Minha sugestão - abra depois de responder</summary>
  <a href="https://www.vaultproject.io/">Vault</a>  
</details>

**Comment by didifive** [comment]
Antes de escrever aqui já cliquei na sugestão, como já vi spoiler só vou aproveitar para implementar no projeto também. Apenas diria que quando se envolve secrets eu acabo remetendo ao KeyVault da Azure já que é a plataforma que utilizo mais.

**Comment by IgorEdu** [comment]
O próprio Docker possui um gerenciador de secrets, porém somente disponível para serviços Swarm.
[Docker secrets](https://docs.docker.com/engine/swarm/secrets/)

Além disso existe a possibilidade de cadastrar as secrets dentro do GitHub e utiliza-las no deploy usando o GitHub Actions.
[GitHub secrets](https://docs.github.com/pt/actions/security-guides/using-secrets-in-github-actions)

Outra opção seria o uso da ferramenta Vault (HashCorp) para fazer a gestão e proteção das secrets
[Vault secrets](https://www.vaultproject.io/)

**Comment by mauromortoni** [comment]
Sugestões 

1. Uso de Variáveis de Ambiente:
   - Armazene credenciais e configurações sensíveis em variáveis de ambiente.
   - Utilize arquivos ".env" e adicione-os ao ".gitignore" para evitar exposição.

2. Separação de Configurações Sensíveis:
   - Mantenha as configurações sensíveis em arquivos separados dos arquivos de configuração gerais.
   - Utilize exemplos de configuração (como "docker-compose.example.yml") sem dados sensíveis.

3. Controle de Acesso:
   - Limite o acesso a informações sensíveis com autenticação forte e controle de permissões.
   - Utilize grupos e roles para gerenciar quem pode acessar quais informações.

4. Uso de Ferramentas de Gerenciamento de Segredos:
   - Considere usar soluções como Docker Secrets, HashiCorp Vault ou AWS Secrets Manager para gerenciar credenciais de forma segura.

5. Criptografia:
   - Utilize criptografia para armazenar informações sensíveis, como senhas (use hashing) e dados de usuários.

6. Auditoria e Monitoramento:
   - Realize auditorias de segurança regulares para identificar e mitigar riscos.
   - Monitore logs para detectar acessos não autorizados.

7. Educação e Conscientização:
   - Treine sua equipe sobre a importância da segurança de dados e melhores práticas para evitar vazamentos.

8. Revisão de Código:
   - Faça revisões de código para garantir que não haja exposição acidental de informações sensíveis.

9. Uso de Repositórios Privados:
   - Mantenha repositórios que contêm informações sensíveis em modo privado.

10. Remoção de Dados Sensíveis do Histórico:
    - Se informações sensíveis forem acidentalmente expostas, use ferramentas para remover esses dados do histórico do Git.

Seguir essas práticas ajudará a proteger suas aplicações e dados contra vazamentos e acessos não autorizados, garantindo a segurança e a conformidade com regulamentos de proteção de dados.
